pkg/driver/node/envprovider/provider_test.go

```diff
@@ -0,0 +1,211 @@
+package envprovider_test
+
+import (
+	"testing"
+
+	"github.com/awslabs/aws-s3-csi-driver/pkg/driver/node/envprovider"
+	"github.com/awslabs/aws-s3-csi-driver/pkg/util/testutil/assert"
+)
+
+func TestGettingRegion(t *testing.T) {
+	testCases := []struct {
+		name             string
+		envRegion        string
+		envDefaultRegion string
+		want             string
+	}{
+		{
+			name:             "both region envs are set",
+			envRegion:        "us-west-1",
+			envDefaultRegion: "us-east-1",
+			want:             "us-west-1",
+		},
+		{
+			name:             "only default region env is set",
+			envRegion:        "",
+			envDefaultRegion: "us-east-1",
+			want:             "us-east-1",
+		},
+		{
+			name:             "no region env is set",
+			envRegion:        "",
+			envDefaultRegion: "",
+			want:             "",
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			t.Setenv("AWS_REGION", testCase.envRegion)
+			t.Setenv("AWS_DEFAULT_REGION", testCase.envDefaultRegion)
+			assert.Equals(t, testCase.want, envprovider.Region())
+		})
+	}
+}
+
+func TestProvidingDefaultEnvironmentVariables(t *testing.T) {
+	testCases := []struct {
+		name string
+		env  map[string]string
+		want []string
+	}{
+		{
+			name: "no env vars set",
+			env:  map[string]string{},
+			want: []string{},
+		},
+		{
+			name: "some allowed env vars set",
+			env: map[string]string{
+				"AWS_REGION":                 "us-west-1",
+				"AWS_DEFAULT_REGION":         "us-east-1",
+				"AWS_STS_REGIONAL_ENDPOINTS": "regional",
+				"AWS_MAX_ATTEMPTS":           "10",
+			},
+			want: []string{
+				"AWS_DEFAULT_REGION=us-east-1",
```

Comment:
Missing test cases where environment variables don't exist

Reply:
It's also missing cases where random environment variables we don't care about aren't passed through

Reply:
The previous test-case `no env vars set` (L52-56) tests when there is no env variable set, and this one includes additional `AWS_MAX_ATTEMPTS=10` which shouldn't be passed. But I can move this to a separate test-case to make it clear.